Client/Frontend/BraveVPN/BraveVPN.swift

```diff
@@ -290,6 +290,34 @@ class BraveVPN {
     /// Connects to Guardian's server to validate locally stored receipt.
     /// Returns true if the receipt expired, false if not or nil if expiration status can't be determined.
     static func validateReceipt(receiptHasExpired: ((Bool?) -> Void)? = nil) {
+        /// A helper struct to deal with unwrapping receipt json and extracting properites we are interested in.
+        struct _Receipt: Decodable {
+            let expireDate: Date
+            let isTrialPeriod: Bool?
+
+            private enum CodingKeys: String, CodingKey {
+                case expiresDateMs = "expires_date_ms"
+                case isTrialPeriod = "is_trial_period"
+            }
+            
+            init(from decoder: Decoder) throws {
+                let container = try decoder.container(keyedBy: CodingKeys.self)
+                let expireDateString = try container.decode(String.self, forKey: .expiresDateMs)
+                if let expireDateTimeIntervalMs = TimeInterval(expireDateString) {
+                    // Expiration date comes in milisecond while the iOS time interval uses seconds.
+                    self.expireDate = Date(timeIntervalSince1970: expireDateTimeIntervalMs / 1000.0)
+                } else {
+                    throw "Casting `expireDateString` failed"
+                }
+                
+                if let isTrialString = try container.decodeIfPresent(String.self, forKey: .isTrialPeriod) {
+                    self.isTrialPeriod = Bool(isTrialString)
+                } else {
+                    self.isTrialPeriod = nil
```

Comment:
Is setting this to nil required? (Can't recall off the top of my head if it is because its a `let` definition)

Reply:
It is, shows this error if I remove it
```
Return from initializer without initializing all stored properties
```